apis/v1alpha1/generator.yaml

```diff
@@ -16,15 +16,60 @@ ignore:
       - RealtimeLogConfig
       - ResponseHeadersPolicy
       - StreamingDistribution
+  field_paths:
+    # NOTE(jaypipes): This is an idempotency token and handled automatically by
+    # the controller; we don't want it as a field on the resource.
```

Comment:
@a-hilaly what field is this not referring to? are we adding something we shouldn't?

Reply:
@acornett21 We are ignore this field. `CallerReference` isn't a field that should be injected in the spec, but rather filled by a random string generated by the controller.

Reply:
From the CF docs:
```
A unique value (for example, a date-time stamp) that ensures that the request can't be replayed.

If the value of CallerReference is new (regardless of the content of the DistributionConfig object), CloudFront creates a new distribution.

If CallerReference is a value that you already sent in a previous request to create a distribution, CloudFront returns a DistributionAlreadyExists error.
```

Reply:
The reason why we want to add this field in the status, is to avoid re-creating distributions if the user updates the desired state. We need to keep track of this value all the time.

Reply:
Ahh Okay, it's all because of CloundFront handles things, so instead of having the field in the CR, we handle it in the controller? Is that correct?

Reply:
Correct. It's instantiated here https://github.com/aws-controllers-k8s/cloudfront-controller/pull/14/files#diff-827b3a739dac32d1033b014758778cdbfc073d45ffa1d5ad6a39113bdbad9dcdR22-R27